src/MyMoneroApi.js

```diff
@@ -0,0 +1,300 @@
+      address: address,
+      api_key: this.apiKey,
+      create_account: true,
+      view_key: privateViewKey
+    })
+    // const asLogin = asObject({
+    //   new_address: asBoolean,
+    //   start_height: asNumber
+    // })
+
+    return result // TODO: Clean this
+  }
+
+  async getTransactions(keys: WalletKeys): Promise<Object[]> {
+    const { address, privateSpendKey, privateViewKey, publicSpendKey } = keys
+    const result = await this.fetchPostMyMonero('get_address_txs', {
+      address,
+      api_key: this.apiKey,
+      view_key: privateViewKey
+    })
+    // asGetAddressTxs = asObject({
+    //   blockchain_height: asNumber,
+    //   scanned_block_height: asNumber,
+    //   scanned_height: asNumber,
+    //   start_height: asNumber,
+    //   total_received: asString,
+    //   transaction_height: asNumber,
+    //   transactions: asArray(asObject({
+    //     coinbase: asBoolean,
+    //     hash: asString,
+    //     height: asNumber,
+    //     id: asNumber,
+    //     mempool: asBoolean,
+    //     mixin: asNumber,
+    //     timestamp: asDate,
+    //     total_received: asString,
+    //     total_sent: asString,
+    //     unlock_time: asNumber
+    //   }))
+    // })
+
+    const parsed = await parserUtils.Parsed_AddressTransactions__async(
+      this.keyImageCache,
+      result,
+      address,
+      privateViewKey,
+      publicSpendKey,
+      privateSpendKey,
+      this.cppBridge
+    )
+    return parsed.serialized_transactions
+  }
+
+  async getAddressInfo(keys: WalletKeys): Promise<BalanceResults> {
+    const { address, privateSpendKey, privateViewKey, publicSpendKey } = keys
+    const result = await this.fetchPostMyMonero('get_address_info', {
+      address,
+      api_key: this.apiKey,
+      view_key: privateViewKey
+    })
+    // const asAddressInfo = asObject({
+    //   blockchain_height: asNumber,
+    //   locked_funds: asString,
+    //   rates: asObject(asNumber),
+    //   scanned_block_height: asNumber,
+    //   scanned_height: asNumber,
+    //   spent_outputs: asArray(
+    //     asObject({
+    //       amount: asString,
+    //       key_image: asString,
+    //       mixin: asNumber,
+    //       out_index: asNumber,
+    //       tx_pub_key: asString
+    //     })
+    //   ),
+    //   start_height: asNumber,
+    //   total_received: asString,
+    //   total_sent: asString,
+    //   transaction_height: asNumber
+    // })
+
+    try {
+      const parsed = await parserUtils.Parsed_AddressInfo__async(
+        this.keyImageCache,
+        result,
+        address,
+        privateViewKey,
+        publicSpendKey,
+        privateSpendKey,
+        this.cppBridge
+      )
+      return {
+        blockHeight: parsed.blockchain_height,
+        totalReceived: parsed.total_received_String,
+        lockedBalance: parsed.locked_balance_String,
+        totalSent: parsed.total_sent_String
+      }
+    } catch (error) {
+      console.log(error)
+    }
+    return {
```

Comment:
This should throw instead of returning safe values so network issues don't result in overwriting existing balance values

Reply:
Concur with Matt. We'd lose cached balance values just due to a network error.

Reply:
Good catch. Not sure how I missed this one!